lib/order.rb

```diff
@@ -1,18 +1,90 @@
+require 'csv'
+require 'awesome_print'
+
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+
     def initialize(id, products)
       @id = id
       @products = products
     end
 
-    def total
-      # TODO: implement total
+
+    def self.all
+      orders = []
+      # "This is a class method."
+      # This is a class method to access the class variabel
+      #returns the csv file to an array with headers
+      #orders_csv = CSV.read('support/orders.csv', 'r')
+      orders_csv = CSV.read("support/orders.csv", 'r',headers: true).to_a
+
+
+      orders_csv.each do |line|
+        orders << Grocery::Order.new(*line)
+      end
```

Comment:
While line 26 creates an `Order`, the products aren't in the right format. An `Order` is supposed to take a hash of products, but here you're passing the raw string full of colons and semicolons from the CSV file. That means that the following code would break

```ruby
order = Grocery::Order.find(7)
order.add_product('tea', 3.50)
```

because `add_product` requires `@products` to be a hash.